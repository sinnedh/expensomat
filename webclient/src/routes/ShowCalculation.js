import React from 'react';
import { connect } from 'react-redux';
import { createExpense, deleteExpense } from '../actions';
import ExpenseForm from '../components/ExpenseForm';
import ExpenseList from '../components/ExpenseList';
import ExpenseMatrix from '../components/ExpenseMatrix';

class ShowCalculation extends React.Component {
  render() {
    return (
      <div>
        <h1>{this.props.name}</h1>
        <p>{this.props.description}</p>
        <h2>Matrix</h2>
        <ExpenseMatrix members={this.props.members} elements={this.props.matrix} />
        <h2>Expenses</h2>
        <ExpenseForm handleSubmit={this.props.onSubmitExpenseForm} members={this.props.members} />
        <ExpenseList onClickDelete={this.props.onClickDelete} expenses={this.props.expenses} />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  token: state.getIn(['application', 'token']),
  expenses: state.get('expenses').toJS(),
  name: state.getIn(['calculations', 'name']),
  description: state.getIn(['calculations', 'description']),
  members: state.getIn(['calculations', 'members']).toJS(),
  matrix: state.getIn(['calculations', 'matrix']).toJS(),
});

const mapDispatchToProps = (dispatch, ownProps) => {
  const token = ownProps.match.params.token

  return {
    onSubmitExpenseForm: (expense) => {
      dispatch(createExpense(token, expense));
    },
    onClickDelete: (event, expense) => {
      event.preventDefault();
      dispatch(deleteExpense(token, expense.id));
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ShowCalculation);
