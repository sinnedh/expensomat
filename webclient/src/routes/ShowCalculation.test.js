import React from 'react'
import { mount, shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import ShowCalculation from './ShowCalculation'

describe('ShowCalculation', () => {
  it('renders properly', () => {
    const initialState = {
      application: { token: 'ABCD1234' },
      calculations: {
        name: 'My calculation',
        description: 'Blabla',
        members: [{name: 'A guy'}, {name: 'Another guy'}]
      },
      expenses: {
        items: [
          {id: 1, description: 'Expense 1', amount: 999, paid_at: '2017-11-11'},
          {id: 2, description: 'Expense 2', amount: 1599, paid_at: '2016-12-08'}
        ]
      }
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<ShowCalculation store={store} />)
    expect(wrapper.dive()).toMatchSnapshot()
  })
})
