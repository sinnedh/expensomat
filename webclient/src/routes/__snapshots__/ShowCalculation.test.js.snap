// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCalculation renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowCalculation
    description="Blabla"
    expenses={
        Object {
            "1": Object {
              "amount": 999,
              "description": "Expense 1",
              "id": 1,
              "paid_at": "2017-11-11",
            },
            "2": Object {
              "amount": 1599,
              "description": "Expense 2",
              "id": 2,
              "paid_at": "2016-12-08",
            },
          }
    }
    match={
        Object {
            "params": Object {
              "token": "ABCD1234",
            },
          }
    }
    matrix={Object {}}
    members={
        Object {
            "1": Object {
              "id": 1,
              "name": "A guy",
            },
            "2": Object {
              "id": 2,
              "name": "Another guy",
            },
          }
    }
    name="My calculation"
    onClickDelete={[Function]}
    onSubmitExpenseForm={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
    }
    token="ABCD1234"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          My calculation
</h1>,
        <p>
          Blabla
</p>,
        <h2>
          Matrix
</h2>,
        <Unknown
          elements={Object {}}
          members={
                    Object {
                              "1": Object {
                                "id": 1,
                                "name": "A guy",
                              },
                              "2": Object {
                                "id": 2,
                                "name": "Another guy",
                              },
                            }
          }
/>,
        <h2>
          Expenses
</h2>,
        <ExpenseForm
          handleSubmit={[Function]}
          members={
                    Object {
                              "1": Object {
                                "id": 1,
                                "name": "A guy",
                              },
                              "2": Object {
                                "id": 2,
                                "name": "Another guy",
                              },
                            }
          }
/>,
        <Unknown
          expenses={
                    Object {
                              "1": Object {
                                "amount": 999,
                                "description": "Expense 1",
                                "id": 1,
                                "paid_at": "2017-11-11",
                              },
                              "2": Object {
                                "amount": 1599,
                                "description": "Expense 2",
                                "id": 2,
                                "paid_at": "2016-12-08",
                              },
                            }
          }
          onClickDelete={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My calculation",
        },
        "ref": null,
        "rendered": "My calculation",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Blabla",
        },
        "ref": null,
        "rendered": "Blabla",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Matrix",
        },
        "ref": null,
        "rendered": "Matrix",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "elements": Object {},
          "members": Object {
            "1": Object {
              "id": 1,
              "name": "A guy",
            },
            "2": Object {
              "id": 2,
              "name": "Another guy",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expenses",
        },
        "ref": null,
        "rendered": "Expenses",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
          "members": Object {
            "1": Object {
              "id": 1,
              "name": "A guy",
            },
            "2": Object {
              "id": 2,
              "name": "Another guy",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "expenses": Object {
            "1": Object {
              "amount": 999,
              "description": "Expense 1",
              "id": 1,
              "paid_at": "2017-11-11",
            },
            "2": Object {
              "amount": 1599,
              "description": "Expense 2",
              "id": 2,
              "paid_at": "2016-12-08",
            },
          },
          "onClickDelete": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            My calculation
</h1>,
          <p>
            Blabla
</p>,
          <h2>
            Matrix
</h2>,
          <Unknown
            elements={Object {}}
            members={
                        Object {
                                    "1": Object {
                                      "id": 1,
                                      "name": "A guy",
                                    },
                                    "2": Object {
                                      "id": 2,
                                      "name": "Another guy",
                                    },
                                  }
            }
/>,
          <h2>
            Expenses
</h2>,
          <ExpenseForm
            handleSubmit={[Function]}
            members={
                        Object {
                                    "1": Object {
                                      "id": 1,
                                      "name": "A guy",
                                    },
                                    "2": Object {
                                      "id": 2,
                                      "name": "Another guy",
                                    },
                                  }
            }
/>,
          <Unknown
            expenses={
                        Object {
                                    "1": Object {
                                      "amount": 999,
                                      "description": "Expense 1",
                                      "id": 1,
                                      "paid_at": "2017-11-11",
                                    },
                                    "2": Object {
                                      "amount": 1599,
                                      "description": "Expense 2",
                                      "id": 2,
                                      "paid_at": "2016-12-08",
                                    },
                                  }
            }
            onClickDelete={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My calculation",
          },
          "ref": null,
          "rendered": "My calculation",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Blabla",
          },
          "ref": null,
          "rendered": "Blabla",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Matrix",
          },
          "ref": null,
          "rendered": "Matrix",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "elements": Object {},
            "members": Object {
              "1": Object {
                "id": 1,
                "name": "A guy",
              },
              "2": Object {
                "id": 2,
                "name": "Another guy",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expenses",
          },
          "ref": null,
          "rendered": "Expenses",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
            "members": Object {
              "1": Object {
                "id": 1,
                "name": "A guy",
              },
              "2": Object {
                "id": 2,
                "name": "Another guy",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "expenses": Object {
              "1": Object {
                "amount": 999,
                "description": "Expense 1",
                "id": 1,
                "paid_at": "2017-11-11",
              },
              "2": Object {
                "amount": 1599,
                "description": "Expense 2",
                "id": 2,
                "paid_at": "2016-12-08",
              },
            },
            "onClickDelete": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
